src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2666,6 +2666,10 @@ void MergeTreeData::removePartsFinally(const MergeTreeData::DataPartsVector & pa
         for (const auto & part : parts)
         {
             part_log_elem.partition_id = part->info.partition_id;
+            {
```

Comment:
We might also need to do add the same handling in writePartLog. which is used in DOWNLOAD_PART event type. https://github.com/ClickHouse/ClickHouse/blob/1263cb24bb3684c10b6829b2c8fbfb9f45e62dac/src/Storages/StorageReplicatedMergeTree.cpp#L4738

Reply:
It is added here: https://github.com/ClickHouse/ClickHouse/pull/70848/files#diff-b7d33a7abda5f3365d5e4fb58ca98b756f6401372b2e2d5eb449a2bd3444fb5eR7935